storage/statedb/cache.go

```diff
@@ -77,7 +86,7 @@ func (cacheType TrieNodeCacheType) ToValid() TrieNodeCacheType {
 
 // NewTrieNodeCache creates one type of any supported trie node caches.
 // NOTE: It returns (nil, nil) when the cache type is CacheTypeLocal and its size is set to zero.
-func NewTrieNodeCache(config TrieNodeCacheConfig) (TrieNodeCache, error) {
+func NewTrieNodeCache(config *TrieNodeCacheConfig) (TrieNodeCache, error) {
```

Comment:
Were there any reason that we used the cache config as a struct, not a pointer? @aidan-kwon @winnie-byun @KimKyungup

Reply:
I put "struct" to omit `nil` checking for `*TrieNodeCacheConfig` and its fields. We can change it to "pointer" if appropriate `nil` checking is added together.

Reply:
@aidan-kwon 73570247525c4bd99d8ae4cd323ac4460730f103 sets default value of TrieNodeCacheConfig if it is nil. I think it would be enough. How do you think?

Reply:
@ehnuje Thank you. I think it's okay. I am gonna add more checking logic in `NewTrieNodeCache` to avoid panic caused by developers' mistakes after this PR is merged.